src/iosrv/vos.c

```diff
@@ -93,8 +94,13 @@ is_sgl_kds_full(struct dss_enum_arg *arg, daos_size_t size)
 	if (arg->sgl_idx < sgl->sg_nr && sgl->sg_nr_out < arg->sgl_idx + 1)
 		sgl->sg_nr_out = arg->sgl_idx + 1;
 
+	if (arg->need_punch)
+		kds_cap = arg->kds_cap - 1; /* extra kds to hold punch epoch */
```

Comment:
We only pack punch entry if it's non-zero.   Should this check take that into account?  Not a huge issue, I suppose but if we don't actually need the extra space, no need to force user to allocate more

Reply:
I thought about it, but since this is only need during rebuild, and we have a lot kds during rebuild, so I did not check non-zero punch, since it needs extra parameter in this function. Though I can add it.